e2e/nsenter.go

```diff
@@ -85,46 +84,17 @@ func extractLogsFromVM(ctx context.Context, t *testing.T, cloud *azureClient, ku
 		mergedCmd := fmt.Sprintf("%s %s", sshCommand, sourceCmd)
 		cmd := append(nsenterCommandArray(), mergedCmd)
 
-		req := kube.typed.CoreV1().RESTClient().Post().Resource("pods").Name(podName).Namespace(defaultNamespace).SubResource("exec")
-
-		option := &corev1.PodExecOptions{
-			Command: cmd,
-			Stdout:  true,
-			Stderr:  true,
-		}
-
-		req.VersionedParams(
-			option,
-			scheme.ParameterCodec,
-		)
-
-		exec, err := remotecommand.NewSPDYExecutor(kube.rest, "POST", req.URL())
-		if err != nil {
-			return nil, err
-		}
-
-		var stdout, stderr bytes.Buffer
-
-		err = exec.StreamWithContext(ctx, remotecommand.StreamOptions{
-			Stdout: &stdout,
-			Stderr: &stderr,
-		})
-
-		t.Log("stdout")
-		t.Log(stdout.String())
-
-		t.Log("stderr")
-		t.Log(stderr.String())
+		t.Logf("executing command on pod %s/%s: %q", defaultNamespace, podName, strings.Join(cmd, " "))
 
+		stdout, stderr, err := execOnPod(ctx, kube, defaultNamespace, podName, cmd)
 		if err != nil {
 			return nil, err
 		}
 
+		checkStdErr(stderr, t)
```

Comment:
this still isn't quite right...on L91 you'll return before logging the error if err != nil (for example, exit 1) and stderr may have contents (script failure output)

Reply:
could either log by default or bubble to caller if err != nil somehow

Reply:
herp